distributed-databases/cloudnative-pg/terraform/cnpg/resources.tf

```diff
@@ -0,0 +1,51 @@
+module "barman_s3_bucket" {
+  source  = "terraform-aws-modules/s3-bucket/aws"
+  version = "~> 3.0"
+
+  bucket = "cnpg-barman-backup-bucket"
+  acl    = "private"
+
+  # For example only - please evaluate for your environment
+  force_destroy = true
+
+  attach_deny_insecure_transport_policy = true
+  attach_require_latest_tls_policy      = true
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+
+  server_side_encryption_configuration = {
+    rule = {
+      apply_server_side_encryption_by_default = {
+        sse_algorithm = "AES256"
+      }
+    }
+  }
+
+  tags = local.tags
+}
+
+#---------------------------------------------------------------
+# IRSA for Barman S3
+#---------------------------------------------------------------
+module "barman_irsa" {
+  source                     = "github.com/aws-ia/terraform-aws-eks-blueprints//modules/irsa?ref=v4.25.0"
```

Comment:
We are going to deprecate the IRSA module from EKS Blueprints . Use this new module to create IRSA. https://github.com/aws-ia/terraform-aws-eks-blueprints-addon/blob/8014b10c9e63e1d30c207ae03b90836d9bf0cbe5/examples/complete/main.tf#L124

Reply:
the link is broken.